hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMDbCheckpointServlet.java

```diff
@@ -83,114 +93,171 @@
+    doNothing().when(responseMock).setContentType("application/x-tgz");
+    doNothing().when(responseMock).setHeader(Matchers.anyString(),
+        Matchers.anyString());
+
+    when(responseMock.getOutputStream()).thenReturn(servletOutputStream);
+
+    omDbCheckpointServletMock.init();
+    long initialCheckpointCount =
+        omMetrics.getDBCheckpointMetrics().getNumCheckpoints();
+
+    omDbCheckpointServletMock.doGet(requestMock, responseMock);
+
+    Assert.assertTrue(tempFile.length() > 0);
+    Assert.assertTrue(
+        omMetrics.getDBCheckpointMetrics().
+            getLastCheckpointCreationTimeTaken() > 0);
+    Assert.assertTrue(
+        omMetrics.getDBCheckpointMetrics().
+            getLastCheckpointStreamingTimeTaken() > 0);
+    Assert.assertTrue(omMetrics.getDBCheckpointMetrics().
+        getNumCheckpoints() > initialCheckpointCount);
+  }
+
+  @Test
+  public void testSpnegoEnabled() throws Exception {
+    cluster.shutdown();
+    conf.setBoolean(OZONE_ACL_ENABLED, true);
+    conf.set(OZONE_ADMINISTRATORS, "");
+    conf.set(OZONE_OM_HTTP_AUTH_TYPE, "kerberos");
+    conf.set(OZONE_RECON_KERBEROS_PRINCIPAL_KEY, "recon/host1@REALM");
+
+    setupCluster();
+
+    final OzoneManager om = cluster.getOzoneManager();
+    Collection<String> allowedUsers = om.getOzoneAdmins(om.getConfiguration());
+    allowedUsers.add("recon");
+
+    doCallRealMethod().when(omDbCheckpointServletMock).initialize(
+        om.getMetadataManager().getStore(),
+        om.getMetrics().getDBCheckpointMetrics(),
+        om.getAclsEnabled(),
+        allowedUsers,
+        om.isSpnegoEnabled());
+
+    omDbCheckpointServletMock.init();
+    omDbCheckpointServletMock.doGet(requestMock, responseMock);
+
+    // Response status should be set to 403 Forbidden since there was no user
+    // principal set in the request
+    verify(responseMock, times(1)).setStatus(HttpServletResponse.SC_FORBIDDEN);
+
+    // Set the principal to DN in request
+    // This should also get denied since only OM and recon
+    // users should be granted access to the servlet
+    Principal userPrincipalMock = mock(Principal.class);
+    when(userPrincipalMock.getName()).thenReturn("dn/localhost@REALM");
+    when(requestMock.getUserPrincipal()).thenReturn(userPrincipalMock);
+
+    omDbCheckpointServletMock.doGet(requestMock, responseMock);
+
+    // Verify that the Response status is set to 403 again for DN user.
+    verify(responseMock, times(2)).setStatus(HttpServletResponse.SC_FORBIDDEN);
+
+    // Now, set the principal to recon in request
+    when(userPrincipalMock.getName()).thenReturn("recon/localhost@REALM");
+
+    when(requestMock.getUserPrincipal()).thenReturn(userPrincipalMock);
+    when(responseMock.getOutputStream()).thenReturn(servletOutputStream);
+
+    omDbCheckpointServletMock.doGet(requestMock, responseMock);
 
+    // Recon user should be able to access the servlet and download the
+    // snapshot
+    Assert.assertTrue(tempFile.length() > 0);
```

Comment:
the test cluster instance needs to be shutdown at the end of the test.

Reply:
Yes, it is handled in the `@AfterClass` method - https://github.com/apache/ozone/pull/2200/files#diff-816030b8fda0e307628b659606cf1d37b2117996c83caab10513b1956f72bc13R128